### PR TITLE
Shrimp plushie

### PR DESCRIPTION
**Pull request by Momo8289**
## About The Pull Request
Adds a cute shrimp plushie, as requested by Slig. It can be found in the kitchen on Selene, Helio, and Pubby, or can be ordered from cargo. It also has a fun little cooking interaction.
Shrimp sprite by Slig.
Screenshot 2024-02-08 202249
## Why It's Good For The Game
Cute plushie, and Slig wanted it.
## Changelog
:cl:
add: Added a new shrimp plushie! It can be found in the kitchen on Helio, Selene, and Pubby, and can be ordered from cargo.
/:cl:

### CODE REVIEW

File: fulp_modules/features/toys/code/shrimp_plush.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		return
+
+	var/obj/item/food/target_food = target
+	if(!findtext(target.name, "fried rice"))
+		return
+
+	RegisterSignal(target_food, COMSIG_FOOD_EATEN, PROC_REF(on_eaten))
+	has_fried = TRUE
+	fried_rice = target_food
+	balloon_alert(user, "fried rice!")
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/shrimp_plush.dm:29]: This is all weird, why doesn't the rice just have a boolean var ``fried_by_shrimp`` or something to give the mood on being eaten, there's no need to hold reference to anything like this.

File: fulp_modules/features/toys/code/shrimp_plush.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/obj/item/toy/plush/shrimp/Destroy()
+	. = ..()
+	if(fried_rice)
+		UnregisterSignal(fried_rice, COMSIG_FOOD_EATEN)
+
+/obj/item/toy/plush/shrimp/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(proximity_flag && has_fried)
+		return
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/shrimp_plush.dm:14]:
```suggestion
		return ..()
```

File: fulp_modules/features/toys/code/shrimp_plush.dm [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+	if(!istype(user.mind.assigned_role, /datum/job/cook))
+		return
+
+	if(!istype(target, /obj/item/food))
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/shrimp_plush.dm:19]: Doesnt this work on any type of food, not just rice?

File: fulp_modules/features/toys/code/shrimp_plush.dm [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	. = ..()
+	if(fried_rice)
+		UnregisterSignal(fried_rice, COMSIG_FOOD_EATEN)
+
+/obj/item/toy/plush/shrimp/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(proximity_flag && has_fried)
+		return
+
+	if(!istype(user.mind.assigned_role, /datum/job/cook))
```
Comment (JohnFulpWillard) [inline fulp_modules/features/toys/code/shrimp_plush.dm:16]: This should instead check the title and compare it to the JOB_COOK define, or even better you can check trait for their skillchip.

File: fulp_modules/features/toys/code/shrimp_plush.dm (hunk 1 above)
Comment (Momo8289) [reply fulp_modules/features/toys/code/shrimp_plush.dm:29]: Didn't want to touch tg code, and there's no actual subtype of food that is fried rice

File: fulp_modules/features/toys/code/shrimp_plush.dm (hunk 3 above)
Comment (Momo8289) [reply fulp_modules/features/toys/code/shrimp_plush.dm:19]: The idea is that it only works on fried rice foods, hence the check on line 23. Checking the name of the food is a little weird ik, but I couldn't really think of another way to do it.

**Comment by Momo8289** [comment]
the failed tests arent my fault

File: fulp_modules/features/toys/code/shrimp_plush.dm (hunk 3 above)
Comment (JohnFulpWillard) [reply fulp_modules/features/toys/code/shrimp_plush.dm:20]: Can someone theoretically just use a pen to rename food to fried rice to make it work?

Comment (Momo8289) [reply fulp_modules/features/toys/code/shrimp_plush.dm:19]: ugh yeah, i'll change it to use the initial value instead

**Comment by SgtHunk** [comment]
as shrimple as that

**Pull request merged**